src/build/env.py

```diff
@@ -6,16 +6,50 @@
 import sysconfig
 import tempfile
 import types
-
-from typing import Dict, Iterable, List, Optional, Sequence, Type
-
+from contextlib import contextmanager
+from datetime import datetime
+from typing import Any, Dict, Iterable, List, Optional, Sequence, Type
 
 if sys.version_info[0] == 2:  # pragma: no cover
     FileExistsError = OSError
 
 _HAS_SYMLINK = None  # type: Optional[bool]
 
 
+DARWIN_CACHE = '~/Library/Caches/pypa/build'
+WINDOWS_CACHE = '~\\AppData\\Local\\pypa\\build\\Cache'
+LINUX_CACHE = '~/.cache/pypa/build'
+
+
+def get_shared_folder():
+    return DARWIN_CACHE if sys.platform == 'darwin' else WINDOWS_CACHE if sys.platform == 'win32' else LINUX_CACHE
```

Comment:
I mean ideally we'd just depend on appdirs as all other tools do, but you were not that keen on new dependencies.

Reply:
> I'd respect `XDG_CACHE_HOME` on Linux, if set, otherwise use `~/.cache` for it.

This proposal is very UNIX friendly but you're alleniating Windows/macOs users. 😢 We can't be platform leaning in a PyPa tool.

Reply:
Sorry, I wasn't clear. What I was trying to say is that we should have a fallback, instead of hardcoding for all platforms.

I additionally suggested to respect `XDG_CACHE_HOME` on Linux, in the hardcoded path.